### PR TITLE
Fix GitHub-15: Fixes race condition between rule sets and security rule

### PR DESCRIPTION
**Pull request by dgajera-crest**
@kyongkim-illumio we have fixed the [bug 15](https://github.com/illumio/terraform-provider-illumio-core/issues/15).

### CODE REVIEW

File: docs/resources/rule_set.md [hunk 1]
```diff
@@ -48,26 +48,6 @@ resource "illumio-core_rule_set" "example" {
       href = "/orgs/1/sec_policy/draft/label_groups/65d0ad0f-329a-4ddc-8919-bd0220051fc7"
     }
   }
-
```
Comment (kyongkim-illumio) [inline docs/resources/rule_set.md:51]: I sent a note to Tim Bardzil to see if it's okay remove support for inline rule creation in rule_set. It's an artificial restriction the PM needs to be aware of.

File: example_policy_workflows/workflow_1/ip_lists.tf [hunk 2]
```diff
@@ -1,11 +1,13 @@
 resource "illumio-core_ip_list" "policy-1-iplist1" {
     name = "policy-1-iplist1"
     description = "Ip list testing."
-
-    dynamic "ip_ranges" {
```
Comment (kyongkim-illumio) [inline example_policy_workflows/workflow_1/ip_lists.tf:5]: Why is this change needed?

File: illumio-core/resource_illumio_rule_set.go [hunk 3]
```diff
@@ -366,202 +354,6 @@ func expandIllumioRuleSetScopes(scopes []interface{}) ([][]*models.RuleSetScope,
 	return sps, &diags
 }
 
-func expandIllumioRuleSetSecurityRules(rules []interface{}) ([]*models.SecurityRule, *diag.Diagnostics) {
```
Comment (kyongkim-illumio) [inline illumio-core/resource_illumio_rule_set.go:369]: Can the code to support inline rules be preserved? I'm not convinced we won't need to support this going forward and I'd like to preserve the flexibility to bring this back if requested.

File: docs/resources/rule_set.md (hunk 1 above)
Comment (cmistry-crest) [reply docs/resources/rule_set.md:51]: Let us know when you get the confirmation so we can go ahead.

File: example_policy_workflows/workflow_1/ip_lists.tf (hunk 2 above)
Comment (cmistry-crest) [reply example_policy_workflows/workflow_1/ip_lists.tf:5]: We also identified the same issue between workload and workload interface. Thus we have to update the workflows accordingly.

File: illumio-core/resource_illumio_rule_set.go (hunk 3 above)
Comment (cmistry-crest) [reply illumio-core/resource_illumio_rule_set.go:369]: We have reverted the uncommented lines, thus preserving the code.

**Comment by cmistry-crest** [comment]
@kyongkim-illumio Can you please merge this PR?

File: docs/resources/rule_set.md (hunk 1 above)
Comment (kyongkim-illumio) [reply docs/resources/rule_set.md:51]: Yes. He's okay with the restriction. In fact, the preferred way of rule writing is as a separate resource per the policy team.

File: illumio-core/resource_illumio_rule_set.go (hunk 3 above)
Comment (kyongkim-illumio) [reply illumio-core/resource_illumio_rule_set.go:369]: Thank you.

File: example_policy_workflows/workflow_1/ip_lists.tf (hunk 2 above)
Comment (kyongkim-illumio) [reply example_policy_workflows/workflow_1/ip_lists.tf:5]: Ahh. Great. Thank you for the due diligence.

File: illumio-core/resource_illumio_rule_set.go [hunk 4]
```diff
@@ -122,6 +122,14 @@ func resourceIllumioRuleSet() *schema.Resource {
 					},
 				},
 			},
+			// "rules": {
```
Comment (kyongkim-illumio) [inline illumio-core/resource_illumio_rule_set.go:125]: It would be good to add a comment indicating why large blocks of code are commented out.

Comment (cmistry-crest) [reply illumio-core/resource_illumio_rule_set.go:129]: @kyongkim-illumio We have added the reasons above each block of code that why it is commented. The PR may now be merged.
